### PR TITLE
JMailHelper:isEmailAddress checked the IPv4 and IPv6 address in the domain

### PR DESCRIPTION
**Pull request by wilsonge**
https://github.com/joomla/joomla-cms/pull/724

Removes the '@' symbol from #724

### CODE REVIEW

**Comment by pe7er** [comment]
Could you please update this PR for the current staging?
Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/3990">issues.joomla.org/joomla-cms/3990</a>.</sub>

**Comment by roland-d** [comment]
Hello @wilsonge 

Thank you for your contribution.

The last comment here was on 28th February. So the question is, Is this issue/pull request still valid?
If the issue is still valid can you please rebase to the current staging or re-do the PR so we are able to test and merge.
If no reply is received within 4 weeks we will close this issue.

Thanks for understanding! <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/3990">issues.joomla.org/joomla-cms/3990</a>.</sub>

**Comment by wilsonge** [comment]
I have absolutely no clue by now xD This was a rehash of #724 we did at the issue tracker code sprint in manchester last year. As the code wasn't originally mine - I honestly can't remember what the aim of it was at this point. It may be valid still?

**Comment by roland-d** [comment]
I think it is still valid as we are now using a PHP native function for checking an IP address. I will get it tested during the PBF tomorrow and see what comes out of it.

**Comment by PeterJTHM** [comment]
Without this patch the validation still returns true on invalid IPv4 adresses and always false for IPv6. The integrated PHP function fixed this problem. It returns false and shows a warning message for invalid adresses.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/3990">issues.joomla.org/joomla-cms/3990</a>.</sub>

**Comment by MPompejus** [comment]
I added to the administrator/com_content/views/articles/tmpl/default.php this code

$result1 = JMailHelper::isEmailAddress("zzz@zzz.zz");
echo "Mail zzz@zzz.zz ($result1)\n";
$result11 = JMailHelper::isEmailAddress("zzz.zzz@zz.zz");
echo "Mail zzz.zzz@zz.zz ($result11)\n";
$result12 = JMailHelper::isEmailAddress("zzz@zzz.zzz.zzz.zz");
echo "Mail zzz@zzz.zzz.zzz.zz ($result12)\n";
$result2 = JMailHelper::isEmailAddress("zzzzz.zz");
echo "Mail zzzzzz.zz ($result2)\n";
$result3 = JMailHelper::isEmailAddress("zzz@.zz");
echo "Mail zzz@.zz ($result3)\n";
$result4 = JMailHelper::isEmailAddress("127.0.0.1");
echo "IPv4 127.0.0.1 ($result4)\n";
$result5 = JMailHelper::isEmailAddress("1.2.3.4.5");
echo "IPv4 1.2.3.4.5 ($result5)\n";
$result6 = JMailHelper::isEmailAddress("::1");
echo "IPv6 ::1 ($result6)\n";
$result7 = JMailHelper::isEmailAddress("2001:0db8:85a3:08d3:1319:8a2e:0370:7344");
echo "IPv6 2001:0db8:85a3:08d3:1319:8a2e:0370:7344 ($result7)\n";

before patch result1, result11, result12 is 1(true) other 0(false)
after patch result1, result11, result12 also produce an "Warning: inet_pton(): Unrecognized address zz.zz in C:\xampp\htdocs\bugtesting\libraries\joomla\mail\helper.php on line "

so i replaced zzz@zzz.zz in result1 with zzz@gmx.de which is an Mailprovider, but also the warning "Warning: inet_pton(): Unrecognized address gmx.de in C:\xampp\htdocs\bugtesting\libraries\joomla\mail\helper.php on line"<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/3990">issues.joomla.org/joomla-cms/3990</a>.</sub>

**Comment by Flow87** [comment]
Patchtest unsuccsessful.
Having the same result as the previous poster @MPompejus 
Left a screenshot below to see what happens.

File: libraries/joomla/mail/helper.php [hunk 1]
```diff
@@ -136,10 +136,8 @@ public static function isEmailAddress($email)
 			return false;
 		}
 
-		// No problem if the domain looks like an IP address, ish
-		$regex = '/^[0-9\.]+$/';
-
-		if (preg_match($regex, $domain))
+		// No problem if the domain looks like an IPv4 and IPv6 address
+		if (inet_pton($domain))
```
Comment (MPompejus) [inline libraries/joomla/mail/helper.php]: Hello i tried the patch for Mail and IP adresse Patch with inet_pton and it brings me Warnings with Mails but don't regongnize IP adresses. 

But simple added this code to administrator/components/com_content/views/articles/tmpl/default.php

$email_a = 'joe@example.com';
$email_b = 'bogus';
$ip_a = '127.0.0.1';
$ip_b = '42.42';

$all = array($email_a, $email_b, $ip_a, $ip_b, "::1", "localhost");

if (filter_var($email_a, FILTER_VALIDATE_EMAIL)) {
	echo "This ($email_a) email address is considered valid.\n";
}
if (filter_var($email_b, FILTER_VALIDATE_EMAIL)) {
	echo "This ($email_b) email address is considered valid.\n";
}

if (filter_var($ip_a, FILTER_VALIDATE_IP)) {
	echo "This ($ip_a) IP address is considered valid.\n";
}
if (filter_var($ip_b, FILTER_VALIDATE_IP)) {
	echo "This ($ip_b) IP address is considered valid.\n";
}

foreach($all as $test)
	if (filter_var($test, FILTER_VALIDATE_EMAIL)) {
		echo "This ($test) email address is considered valid.\n";
	}
	elseif (filter_var($test, FILTER_VALIDATE_IP)) {
		echo "This ($test) IP address is considered valid.\n";
	}
	else
	{
		echo "This ($test) is not considered valid IP or Email.\n";
	}

and it worked and recognized the Mail and IPv4 and IPv6 so as you see an possible solution could be to change Mailhelper 

if (filter_var($email, FILTER_VALIDATE_EMAIL)) {
		return true;
}
elseif (filter_var($email, FILTER_VALIDATE_IP)) {
		return true;
}
else	{
		return false;
}

**Comment by roland-d** [comment]
Hello @wilsonge 

Thank you for your contribution.

The last comment here was on 21st August. So the question is, Is this issue/pull request still valid?
if so please provide clear test instructions to be able to test / reproduce this issue.
If no reply is received within 4 weeks we will close this issue.

Thanks for understanding!

**Comment by wilsonge** [comment]
You were the one who told me it was still valid last time :p you tell me

**Comment by roland-d** [comment]
@wilsonge It's valid but the unsuccessful tests must be investigated.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/3990">issues.joomla.org/joomla-cms/3990</a>.</sub>

**Comment by brianteeman** [comment]
@wilsonge what is the ststus of this?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/3990">issues.joomla.org/joomla-cms/3990</a>.</sub>

**Comment by wilsonge** [comment]
Closing. I don't have the time to look into this. If someone wants they can pick it up.

**Pull request closed without merging**